### PR TITLE
Fixes #3924: 65K methods limit

### LINKED ISSUE #3924: Exceeding the 64k method limit
The `feature/gravatar-upload-okhttp` branch (at https://github.com/wordpress-mobile/WordPress-Android/commit/30fa9714d2072d1f635893e1ef97eecfdc65372f) manages to exceed the 64k method limit.

We need to decrease the app's current method count.

### PR DESCRIPTION
**Pull request by maxme**
Fixes #3924: 65K methods limit

1. Minifier support added in 27ea264
1. Multi dex support added for all builds (but "ignored" in release builds) in 4335b23

Vanilla debug method count: 77870
Vanilla release method count: 45550

Notes:
* This is merged with #4236 changes.
* I also added the Kotlin dependency back (emailchecker2) in 392dba7
* I bumped the gms version in 5ef3ee6 because it's needed for targetSdk 24

### CODE REVIEW

**Comment by maxme** [comment]
Travis is failing to run the lint step because platform-tools installed there is v23.0.1 while we need v24-rc3.

**Comment by maxme** [comment]
> Travis is failing to run the lint step because platform-tools installed there is v23.0.1 while we need v24-rc3.

Found a workaround: https://github.com/wordpress-mobile/WordPress-Android/blob/ebb01b8e2d16e0a067d947b9b052bbe24ebaa628/.travis.yml#L32

**Comment by maxme** [comment]
And I confirm it fixes https://code.google.com/p/android/issues/detail?id=78377

File: WordPress/src/main/java/org/wordpress/android/WordPress.java [hunk 1]
```diff
@@ -109,8 +111,13 @@
     private static Context mContext;
     private static BitmapLruCache mBitmapCache;
 
+    protected void attachBaseContext(Context base) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/WordPress.java]: This is already handled by `MultiDexApplication` (in fact, it's the only thing `MultiDexApplication` adds to `Application`).

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/WordPress.java]: Ah good catch, I misread the doc, I'll remove this `attachBaseContext`.

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/WordPress.java]: fixed in 7d681cb

**Comment by nbradbury** [comment]
Looks good :shipit: and 🙏 !

**Pull request merged**